File: 2022 Challenge/1. Engine Diagnostics/engine-diagnostics.py
import os

filename = "engine-diagnostics.txt"
here = os.path.dirname(os.path.abspath(__file__))
filepath = os.path.join(here, filename)

with open(filepath, "r") as f:
    content = [int(x) for x in f.read().splitlines()]

total = 0
for i in range(60, len(content)):
    r_av = sum(content[i-60:i])/60
    total += 1 if r_av < 1500 or r_av > 1600 else 0
print(total) # Answer: 6248

File: 2022 Challenge/1. Engine Diagnostics/engine-diagnostics.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main(){
    ifstream file("engine-diagnostics.txt");
    string line;
    int content[86400], i = 0, r_av = 0;
    while (getline(file, line)){
        content[i] = stoi(line);
        if (++i >= 60) {
            int sum = 0;
            for (int j = i-60; j < i; j++){
                sum += content[j];
            }
            if (sum < 90000 || sum > 96000) {
                ++r_av;
            }
        }
    }
    file.close();
    cout << r_av; // Answer: 6248
}
